sdk/identity/azure-identity/azure/identity/_internal/interactive.py

```diff
@@ -57,16 +57,27 @@ def _build_auth_record(response):
             # MSAL uses the subject claim as home_account_id when the STS doesn't provide client_info
             home_account_id = id_token["sub"]
 
+        # "iss" is the URL of the issuing tenant e.g. https://authority/tenant
+        issuer = six.moves.urllib_parse.urlparse(id_token["iss"])
```

Comment:
Do we need to check if "iss" in id_token?

Reply:
No. It should always be present. If it isn't, this method can't succeed.

Reply:
Yes. I think the question is the error message. We prefer KeyError: 'iss' or something like "You can to specify 'issuer' to make it work."?

Reply:
And same for other keys.

Reply:
Never mind. After I expand to see more code, realize it is only in try catch.